mutation/mutation_compactor.hh

```diff
@@ -683,6 +691,12 @@ public:
     }
 
     stop_iteration consume(clustering_row&& cr) {
+        if constexpr (SSTableCompaction == compact_for_sstables::yes) {
```

Comment:
twcs uses compacting_reader, which bypasses this compact_mutation_v2 wrapper. so let's move it to compact_mutation_state.